slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala

```diff
@@ -136,6 +137,28 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
     } yield ()
   }
 
+  def testInsertOrUpdatePlainWithFuncDefinedPK: DBIOAction[Unit, _, _] = {
+    //FIXME remove this after fixed checkInsert issue
+    if (tdb.profile.isInstanceOf[DerbyProfile]) return DBIO.successful(())
```

Comment:
so the way we handle this usually is capabilities.

Here is where you can define an appropriate one

https://github.com/slick/slick/blob/cada4176b2095563f113da540cc79eea0c7a39c3/slick/src/main/scala/slick/jdbc/JdbcCapabilities.scala#L15

here is where you exclude it from Derby to signal Derby does not support it

https://github.com/slick/slick/blob/6b59d0fc228bc4262491d0cf7110eaac0e0d8a7f/slick/src/main/scala/slick/jdbc/DerbyProfile.scala#L77

Here is how you can check for it in a test:

https://github.com/slick/slick/blob/898cb1871c610cea72027b266494f5bd83f770c9/slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala#L82

The nice thing is that our docs list all of the limitations: http://slick.lightbend.com/doc/3.1.1/supported-databases.html (the list is generated automatically)

Would you mind adding one for this?